install.sh

```diff
@@ -198,7 +205,7 @@ if [ -z ${FC+x} ] || [ -z ${CC+x} ] || [ -z ${CXX+x} ] || [ -z ${PKG_CONFIG+x} ]
     exit_if_user_declines "brew"
 
     if ! command -v curl > /dev/null 2>&1; then
-      echo "No download mechanism found. Please install curl and rerun ./install.sh"
+      echo "curl not found. Please install curl, ensure it is in your PATH, and rerun ./install.sh"
       exit 1
     fi
```

Comment:
We agreed in our meeting that (just like `git`) `curl` is also always required (due to GASNet), so this entire block should also be promoted out of this HomeBrew branch up to an earlier point where it always runs (probably right beside where you added the `git` check).

Similarly `curl` should also be added to the dependency message.